driver/src/contracts/stablex_contract.rs

```diff
@@ -161,19 +161,21 @@ fn parse_auction_data(packed_auction_bytes: Vec<u8>, index: U256) -> (AccountSta
     (account_state, relevant_orders)
 }
 
-fn encode_prices_for_contract(price_vector: Vec<u128>) -> (Vec<U128>, Vec<u64>) {
-    // Representing the solution's price vector more compactly as:
-    // sorted_touched_token_ids, non_zero_prices which are logically bound by index.
-    // Example solution.prices = [3, 0, 1] will be transformed into [0, 2], [3, 1]
-    let mut prices = vec![];
-    let mut ordered_token_ids = vec![];
-    for (token_id, price) in price_vector.into_iter().enumerate().skip(1) {
-        if price > 0 {
-            prices.push(U128::from(price));
-            ordered_token_ids.push(token_id as u64);
-        }
-    }
-    (prices, ordered_token_ids)
+fn encode_prices_for_contract(price_map: HashMap<u16, u128>) -> (Vec<U128>, Vec<u64>) {
+    // Representing the solution's price vector as:
+    // sorted_touched_token_ids, non_zero_prices (excluding price at token with id 0)
+    let mut token_ids: Vec<u16> = price_map
+        .keys()
+        .copied()
+        .filter(|t| *t > 0 && price_map[t] > 0)
+        .collect();
+    token_ids.sort_unstable();
+    let prices = token_ids
+        .iter()
+        .filter(|token_id| price_map[*token_id] > 0)
```

Comment:
Haven't you already checked that above?

Reply:
Yes, I suppose I have. I actually wrote this line before the one above. I guess this was left over. Good catch.